org.uqbar.project.wollok.tests/src/org/uqbar/project/wollok/tests/xpect/validations/mixins/MixinInstantiation.wlk.xt

```diff
@@ -0,0 +1,91 @@
+/* XPECT_SETUP org.uqbar.project.wollok.tests.xpect.WollokXPectTest END_SETUP */
+
+mixin M1 {
+   method m() {
+   	  this.doIt()      // expected abstract method
+   }
+   method doIt()
+}
+
+class C mixed with M1 {} // should be considered abstract
+
+class D inherits C {}
+
+
+mixin M2 {}
+
+class E inherits D mixed with M2 {}
+
+mixin M3 {
+	method blah() { this.sarasa() }
+	method sarasa()
+}
+
+class F inherits E mixed with M3 {}
+
+mixin DoIt {
+	method doIt() {}
+}
+
+class G inherits E mixed with DoIt {}
+
+class DoItClass {
+	method doIt() {}
+}
+
+class H inherits DoItClass mixed with M2 {}
+
+class I mixed with M2 {
+	method doIt() {}
+}
+
+object p {
+	method run() {
+		// XPECT errors --> "Must implement inherited abstract methods: M1.doIt()" at "C"
+		val c = new C()
+		console.println(c)
+		
+		// XPECT errors --> "Must implement inherited abstract methods: M1.doIt()" at "D"
```

Comment:
Why not replace 'M1.doIt()' with 'doIt() (required by M1)

I like the term "required" as more natural dan "abstract".